### PR TITLE
Added Only_instructor_view switch for seating

### PR DESCRIPTION
**Pull request by VictorLi718**
image
image
I add a Only_instructor_view checkbox for seating here. Once selected and you choose any seating assignment, only the instructors will see the seating map(with a short sentence Visible only to instructors).

### CODE REVIEW

**Comment by VictorLi718** [comment]
One thing you have to know is this feature requires another default value in course_config. Should add "seating_only_for_instructor=false" in the ".ini" file for every course

**Comment by bmcutler** [comment]
Victor and I talked today about adding a migration for the config .ini file
(still needs to be done)

File: site/app/models/Config.php [hunk 1]
```diff
@@ -175,8 +176,11 @@ class Config extends AbstractModel {
     protected $regrade_enabled;
     /** @property @var string */
     protected $regrade_message;
+    /** @property @var bool*/
+    protected $seating_only_for_instructor;
     /** @property @var string|null */
     protected $room_seating_gradeable_id;
+
```
Comment (holzbh) [inline site/app/models/Config.php]: Extra newline is unnecessary

**Comment by VictorLi718** [comment]
About the instructor_seating_view, I didn't change anything. Maybe just the changes in the master branch. We should dismiss those changes..?

File: site/app/views/NavigationView.php [hunk 2]
```diff
@@ -100,6 +100,8 @@ public function showGradeables($sections_to_list, $graded_gradeables, array $sub
         // DISPLAY ROOM SEATING (used to display room seating assignments)
         // ======================================================================================
         $display_room_seating = $this->core->getConfig()->displayRoomSeating();
+        $seating_only_for_instructor = $this->core->getConfig()->isSeatingOnlyForInstructor();
+        if(!$this->core->getUser()->accessAdmin() and $seating_only_for_instructor) $display_room_seating = false;
```
Comment (MasterOdin) [inline site/app/views/NavigationView.php]: Combine line 102 and 104 into just a single if statement branch so it's clearer what's happening:
```
if ($seating_only_for_instructor && !$this->core->getUser()->accessAdmin()) {
    $display_room_seating = false;
}
else {
    $display_room_seating = $this->core->getConfig()->displayRoomSeating();
}
```

**Comment by bmcutler** [comment]
@VictorLi718 please address feedback

**Comment by VictorLi718** [comment]
Should be clearer now. And if the migration is fine, I think we can merge it.

File: migration/migrations/course/20180802223149_config_iris_etc.py [hunk 3]
```diff
@@ -25,6 +25,8 @@ def up(config, conn, semester, course):
         if not config.has_option('course_details','private_repository'):
             config.set('course_details','private_repository','""')
         # the newest default
+        if not config.has_option('course_details','seating_only_for_instructor'):
+            config.set('course_details','seating_only_for_instructor','false')
```
Comment (MasterOdin) [inline migration/migrations/course/20180802223149_config_iris_etc.py]: You have to add to this a new migration. Older migrations have already been applied and editing them won't have any affect on existing installations, only new ones.

Comment (VictorLi718) [reply migration/migrations/course/20180802223149_config_iris_etc.py]: I don't know how to do that... Could you please take a look and help?

**Comment by MasterOdin** [comment]
@bmcutler the migration is setup and ready, but I did not test the feature as I'm not really sure the steps here as I've never used RainbowGrades.

**Pull request merged**